plugins/modules/ndo_tenant_mld_snooping_policy.py

```diff
@@ -0,0 +1,316 @@
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    admin_state = module.params.get("admin_state")
+    fast_leave_control = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("fast_leave_control"))
+    querier_control = ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP.get(module.params.get("querier_control"))
+    querier_version = module.params.get("querier_version")
+    query_interval = module.params.get("query_interval")
+    query_response_interval = module.params.get("query_response_interval")
+    last_member_query_interval = module.params.get("last_member_query_interval")
+    start_query_interval = module.params.get("start_query_interval")
+    start_query_count = module.params.get("start_query_count")
+    state = module.params.get("state")
+
+    template_object = MSOTemplate(mso, "tenant", template)
+    template_object.validate_template("tenantPolicy")
+
+    mld_snooping_policies = template_object.template.get("tenantPolicyTemplate", {}).get("template", {}).get("mldSnoopPolicies", [])
+    object_description = "MLD Snooping Policy"
+
+    if state in ["query", "absent"] and mld_snooping_policies == []:
+        mso.exit_json()
+    elif state == "query" and not (name or uuid):
+        mso.existing = mld_snooping_policies
+    elif mld_snooping_policies and (name or uuid):
+        match = template_object.get_object_by_key_value_pairs(
+            object_description, mld_snooping_policies, [KVPair("uuid", uuid) if uuid else KVPair("name", name)]
+        )
+        if match:
+            mld_snooping_policy_attrs_path = "/tenantPolicyTemplate/template/mldSnoopPolicies/{0}".format(match.index)
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+
+    ops = []
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="{0} with the UUID: '{1}' not found".format(object_description, uuid))
+
+        mso_values = dict(
+            name=name,
+            description=description,
+            enableAdminState=admin_state,
+            enableFastLeaveControl=fast_leave_control,
+            enableQuerierControl=querier_control,
+            mldQuerierVersion=querier_version,
+            queryInterval=query_interval,
+            queryResponseInterval=query_response_interval,
+            lastMemberQueryInterval=last_member_query_interval,
+            startQueryInterval=start_query_interval,
+            startQueryCount=start_query_count,
+        )
+
+        if mso.existing:
+            proposed_payload = copy.deepcopy(match.details)
+            append_update_ops_data(ops, proposed_payload, mld_snooping_policy_attrs_path, mso_values)
+            mso.sanitize(proposed_payload, collate=True)
+        else:
+            mso.sanitize(mso_values)
```

Comment:
```suggestion
            mso.sanitize(mso_values, collate=True)
```
Do you need to use `collate=True` here? Will sanitize still remove `None` dictionary keys from the payload without it?

Reply:
Not needed, have removed for you. In this case for the existing there are no none values because it comes from existing so makes no impact on the output. We could argue that we do want to know when the value in existing is actually said to None. I think gaspard had some of these scenarios.

Reply:
the None will always be removed on sanitise

# Add everything else
                if updates.get(key) is not None:
                    self.sent[key] = updates.get(key)